app/services/taxonomy/update_taxon.rb

```diff
@@ -19,10 +19,13 @@ def publish(validate: true)
       end
 
       Services.publishing_api.put_content(content_id, payload)
-      Services.publishing_api.patch_links(
-        content_id,
-        links: { parent_taxons: Array(parent) }
-      )
+
+      if parent.present?
```

Comment:
This class handles the user's form submission, so if there is no parent taxon submitted in the UI then `taxon.parent` will be an empty string.